### PR TITLE
Add hosted API docs

### PR DESCRIPTION
**Pull request by nickrobinson251**
- Since we have decent docstrings, we should probably make these API docs available
- For everything else, I've tried to just point people to the ChainRules docs, rather than repeat info here.
- Hopefully will appear at www.juliadiff.org/ChainRulesTestUtils.jl/preview-PR19
---
- Added a `DOCUMENTER_KEY` variable to travis: https://travis-ci.com/JuliaDiff/ChainRulesTestUtils.jl/settings
- Added a deploy key to the repo: https://github.com/JuliaDiff/ChainRulesTestUtils.jl/settings/keys

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# ChainRulesTestUtils
+
+[![Travis](https://travis-ci.org/JuliaDiff/ChainRulesTestUtils.jl.svg?branch=master)](https://travis-ci.org/JuliaDiff/ChainRulesTestUtils.jl)
+[![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
+
+
+[ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl) helps you test [`ChainRulesCore.frule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) and [`ChainRulesCore.rrule`](http://www.juliadiff.org/ChainRulesCore.jl/dev/api.html) methods, when adding rules for your functions in your own packages.
```
Comment (mattBrzezinski) [inline docs/src/index.md]:
```suggestion
[ChainRulesTestUtils.jl](https://github.com/JuliaDiff/ChainRulesTestUtils.jl) helps you test [`ChainRulesCore.frule`]http://www.juliadiff.org/ChainRulesCore.jl/stable/api.html#ChainRulesCore.frule-Tuple{Any,Vararg{Any,N}%20where%20N}) and [`ChainRulesCore.rrule`]http://www.juliadiff.org/ChainRulesCore.jl/stable/api.html#ChainRulesCore.rrule-Tuple{Any,Vararg{Any,N}%20where%20N}) methods, when adding rules for your functions in your own packages.
```

Comment (nickrobinson251) [reply docs/src/index.md]: I purposefully didn't link the exact path, as it's too liable to change, whereas frule and rrule are likely to always be easy to find on that page

File: docs/make.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+using ChainRulesTestUtils
+using Documenter
+
+@show ENV
```
[the remark this answers is not in the document]
Comment (nickrobinson251) [reply docs/make.jl]: I just copied ChainRulesCore. I presume it's proved useful in the past... Maybe @oxinabox knows?

[the remark this answers is not in the document]
Comment (mattBrzezinski) [reply docs/make.jl]: Ah! It looks like it's an artifact from debugging or something, I don't think it's really necessary to keep around? If it's not should also remove it in `ChainRulesCore` as well.

**Comment by nickrobinson251** [comment]
Idk if this is working... Or how to test it is working (other than merging and seeing if docs build and deploy) -- anyone know?

**Comment by mattBrzezinski** [comment]
> Idk if this is working... Or how to test it is working (other than merging and seeing if docs build and deploy) -- anyone know?

I don't have any experience with Documenter, but can we not just do `julia make.jl` and check what has been built?

**Comment by nickrobinson251** [comment]
Or works locally. I'm just not sure if CI deploy stuff is all good

File: docs/make.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (oxinabox) [reply docs/make.jl]: It is not required. It is an artifact from debugging.
I knew it was there, and didn't bother to remove it as I thought i might need to debug again.

**Comment by oxinabox** [comment]
> Idk if this is working... Or how to test it is working (other than merging and seeing if docs build and deploy) -- anyone know?

I always test docs by pushing to master.
Its the only way I know how.

**Comment by mattBrzezinski** [comment]
I’m happy just sending it with this, maybe just remove the @show line and squash a couple commits and I think it’s good to go!

**Pull request merged**